pkg/pc/control/control.go

```diff
@@ -42,7 +44,26 @@ func NewPodCluster(
 	return pc
 }
 
+func NewPodClusterFromID(
+	ID fields.ID,
+	session kp.Session,
+	pcStore pcstore.Store,
+) *PodCluster {
+	pc := &PodCluster{}
+	pc.session = session
+	pc.pcStore = pcStore
+	pc.ID = ID
+	return pc
+}
+
 func (pccontrol *PodCluster) All() ([]fields.PodCluster, error) {
+	if pccontrol.ID != "" {
+		pc, err := pccontrol.pcStore.Get(pccontrol.ID)
+		if err != nil {
+			return nil, err
+		}
+		return []fields.PodCluster{pc}, err
```

Comment:
maybe clearer to say `nil` instead of `err` here since if we get here `err` is `nil`